src/Algebra/Graph/Labelled/AdjacencyMap/Algorithm/Dijkstra.hs

```diff
@@ -0,0 +1,109 @@
+module Algebra.Graph.Labelled.AdjacencyMap.Algorithm.Dijkstra where
+
+import Data.Map.Strict (Map, (!))
+import Algebra.Graph.Label (Semiring(..), (<+>), zero)
+import qualified Data.Map.Strict as Map
+import Algebra.Graph.Labelled.AdjacencyMap.Internal (AdjacencyMap(..))
+
+-- | Psuedocode for Generic-Single-Source-Shortest-Distance (G,s)
+-- for i <- 1 to Number of Vertices:
+--   do d[i] <- r[i] <- 0 
+-- d[s] ← r[s] ← 1 
+-- q ← {s} 
+-- while q != []
+--   do p <- head(q)
+--      Dequeue(q)
+--      r0 <- r[p]
+--      r[p] <- 0
+--      for each e in E[p]
+--        do if d[n[e]] != d[n[e]] <+> (r0 <.> w[e])
+--             then d[n[e]] <- d[n[e]] <+> (r0 <.> w[e])
+--                  r[n[e]] <- r[n[e]] <+> (r0 <.> w[e])
+--                  if n[e] not in q
+--                    then Enqueue(q, n[e])
+-- d[s] ← 1
+--
+-- Convensions used:
+-- if e is a -> b
+-- n[e] = b
+-- p[e] = a
+--
+-- if p is any vertex
+-- E[p] is the edge set of p
+--
+-- if e is any edge
+-- w[e] is the weight of the edge e
+--
+-- Brief:
+-- We use a queue q to maintain the set of vertices whose leaving edges are to be relaxed. q is initialized to [source].
+-- We maintain 2 attributes, namely, `d` and `r`.
+-- d[p] an estimate of the shortest distance from s to p
+-- r[p] is the total weight added to d[q] since the last time p was extracted for q.
+-- We update d and r till the queue is empty.
+
+dijkstra ::
```

Comment:
The documentation misses examples with corresponding tests, see an example here:

https://hackage.haskell.org/package/algebraic-graphs-0.3/docs/Algebra-Graph-AdjacencyMap-Algorithm.html#v:dfs

Also, what are the time/memory complexity of your implementation?